phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -883,6 +903,14 @@ public void readFields(DataInput input) throws IOException {
         dataEmptyKeyValueCF = Bytes.readByteArray(input);
         emptyKeyValueCFPtr = new ImmutableBytesPtr(Bytes.readByteArray(input));
         
+        int numIndexedExpressions = WritableUtils.readVInt(input);
```

Comment:
This will likely be a b/w compat issue (as old clients wouldn't serialize this. We may need to do something somewhat hacky, like negate an already serialized value as a flag that this index maintainer may contain expressions.